frontend/web/components/AuditLog.tsx

```diff
@@ -96,17 +103,47 @@ const AuditLog: FC<AuditLogType> = (props) => {
     })
     const colour = index === -1 ? 0 : index
     let link: ReactNode = null
-    if (
-      related_object_uuid &&
-      related_object_type === 'EF_VERSION' &&
-      environment
-    ) {
+    const date = moment(created_date)
+    const isVersionEvent =
+      related_object_uuid && related_object_type === 'EF_VERSION' && environment
+    const versionLimitDays = subscriptionMeta?.feature_history_visibility_days
+    const isOverLimit = isVersionEvent
+      ? isVersionOverLimit(versionLimitDays, created_date)
+      : false
+    const VersionButton = (
+      <Button disabled={isOverLimit} theme='text'>
+        View version
+      </Button>
+    )
+    if (isVersionEvent) {
       link = (
-        <Link
-          to={`/project/${project.id}/environment/${environment.api_key}/history/${related_object_uuid}/`}
+        <Tooltip
+          title={
+            <div className='d-flex gap-2'>
+              {isOverLimit ? (
+                VersionButton
+              ) : (
+                <Link
+                  to={`/project/${project.id}/environment/${environment.api_key}/history/${related_object_uuid}/`}
+                >
+                  {VersionButton}
+                </Link>
+              )}
+              <PlanBasedBanner
```

Comment:
I know it's not related to this PR, but this button has too low contrast in dark mode:

![image](https://github.com/user-attachments/assets/a94c734e-237b-4468-9fb3-c90d27d29811)